pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -543,3 +567,20 @@ func constructFilesSecret(driver string, config map[string]interface{}) (*corev1
 	}
 	return nil, nil
 }
+
+func shouldForceRemoval(job *batchv1.Job, d data.Object) bool {
+	var createJobCondition summary.Condition
+	for _, c := range summary.GetUnstructuredConditions(d) {
+		if c.Type() == createJobConditionType {
+			createJobCondition = c
+			break
+		}
+	}
+
+	if createJobCondition.Reason() != string(errors.CreateMachineError) &&
+		d.Map("metadata", "annotations")[forceRemoveMachineAnn] != "true" {
```

Comment:
nit: should we `strings.ToLower` the annotation value?

Reply:
I will do this.